src/api/dynamic_data.c

```diff
@@ -0,0 +1,829 @@
+// Copyright 2022 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <assert.h>
+#include <stdlib.h>
+
+#include <rcutils/allocator.h>
+#include <rcutils/error_handling.h>
+#include <rcutils/types/rcutils_ret.h>
+#include <rcutils/types/uint8_array.h>
+#include <rosidl_dynamic_typesupport/api/dynamic_data.h>
+#include <rosidl_dynamic_typesupport/macros.h>
+#include <rosidl_dynamic_typesupport/types.h>
+#include <rosidl_dynamic_typesupport/uchar.h>
```

Comment:
nitpick: I think according to our style guide this should be like this:

```suggestion
#include <rosidl_dynamic_typesupport/api/dynamic_data.h>

#include <assert.h>
#include <stdlib.h>

#include <rcutils/allocator.h>
#include <rcutils/error_handling.h>
#include <rcutils/types/rcutils_ret.h>
#include <rcutils/types/uint8_array.h>

#include <rosidl_dynamic_typesupport/macros.h>
#include <rosidl_dynamic_typesupport/types.h>
#include <rosidl_dynamic_typesupport/uchar.h>
```